tensorbay/apps/sextant.py

```diff
@@ -0,0 +1,122 @@
+#
+
+# flake8: noqa: F401
+# pylint: disable=no-self-use
+# pylint: disable=too-many-arguments
+# pylint: disable=missing-function-docstring
+
+"""Interact with sextant app at graviti marketplace."""
+
+from typing import Any, Dict, List, Optional
+from urllib.parse import urljoin
+
+from tensorbay.client.requests import Client
+
+
+class Evaluation:
+    """This class defines :class:`Evaluation`.
+
+    Arguments:
+        evaluation_id: Evaluation ID.
+        benchmark: The :class:`Benchmark`.
+
+    """
+
+    def __init__(self, evaluation_id: str, benchmark: "Benchmark") -> None:
+        pass
+
+    def result(self) -> Dict[str, Any]:
+        """Get the result of the evaluation.
+
+        Returns:
+            The result dict of the evaluation.
+
+        """
+        return {"a": {}}
+
+    def status(self) -> str:
+        pass
+
+
+class Benchmark:
+    """This class defines :class:`Benchmark`.
+
+    Arguments:
+        name: name of the Benchmark.
+        dataset_id: ID of the dataset which this benchmark based.
+        commit_id: ID of the commit which used as the evaluation benchmark.
+        benchmark_id: Benchmark ID.
+        client: The :class:`SextantClient`.
+        iou: The ioU threshold.
+        customized_metrics: https url of the github repository.
+
+    Raises:
+        ValueError: When create the benchmark based on iou and customized_metrics.
+
+    """
+
+    def __init__(
+        self,
+        name: str,
+        dataset_id: str,
+        commit_id: str,
+        benchmark_id: str,
+        client: "SextantClient",
+        categories: Optional[List[str]] = None,
+        iou: Optional[float] = None,
+        customized_metrics: Optional[str] = None,
+    ) -> None:
+        pass
+
+    def create_evaluation(self, dataset_id: str, commit_id: str) -> Evaluation:
+        """Create a evaluation task.
+
+        Arguments:
+            dataset_id: Id of the needed  evaluation dataset.
+            commit_id: Id of the needed commit.
+
+        Return:
+            The created evaluation instance.
+
+        """
+
+    def list_evaluation(self) -> List[Evaluation]:
+        pass
+
+
+class SextantClient(Client):
+    """This class defines :class:`SextantClient`."""
+
+    def __init__(self, access_key: str, url: str = "") -> None:
+        super().__init__(access_key, url)
+        self._open_api = urljoin(self.gateway_url, "apps/sextant/")
+
+    def create_benchmark(
+        self,
+        name: str,
+        dataset_id: str,
+        commit_id: str,
+        categories: Optional[List[str]] = None,
+        iou: Optional[float] = None,
+        customized_metrics: Optional[str] = None,
+    ) -> Benchmark:
+        """Create a benchmark with the given parameters."""
+
+    def list_benchmark(self) -> List[Benchmark]:
+        """List all benchmarks.
+
+        Return:
+            The List of Benchmark instance.
+
+        """
+
+    def get_benchmark(self, name: str) -> Benchmark:
```

Comment:
Does this method mean benchmark has unique name as identifer?

Reply:
Yes, benchmark has the unique name.